spec/rails_state_machine/model_spec.rb

```diff
@@ -45,4 +45,17 @@ class MyClass < ActiveRecord::Base
 
   end
 
+  it 'does not remove already defined state machines if you include the module again afterwards (BUGFIX)' do
+    class_with_machine = Class.new(ActiveRecord::Base) do
+      include RailsStateMachine::Model
+      state_machine(:foo) {}
+      state_machine(:bar) {}
+      include RailsStateMachine::Model
```

Comment:
From my understanding, the issue which pointed to this bug was in a project where `include` happened before each state machine definition. Right now, the test does not reflect that and the bug might be reintroduced if only `state_machine` resets the `state_machines` registry, not the `include` statement itself.

IMO this is more robust and more explicit:

```suggestion
      include RailsStateMachine::Model
      state_machine(:foo) {}
      include RailsStateMachine::Model
      state_machine(:bar) {}
```

Reply:
Yes, the current bugged behavior is that `include` breaks it. However, the test itself somewhat depends on that while it should not.

Thanks for adjusting.